tests/tracer/test_single_span_sampling_rule.py

```diff
@@ -0,0 +1,59 @@
+from ddtrace.constants import _SINGLE_SPAN_SAMPLING_MAX_PER_SEC
+from ddtrace.constants import _SINGLE_SPAN_SAMPLING_MECHANISM
+from ddtrace.constants import _SINGLE_SPAN_SAMPLING_RATE
+from ddtrace.single_span_sampling_rule import SAMPLING_MECHANISM
+from ddtrace.single_span_sampling_rule import SpanSamplingRule
+
+from ..utils import DummyTracer
+
+
+def assert_sampling_decision_tags(span, sample_rate=1.0, mechanism=SAMPLING_MECHANISM, limit=None):
+    assert span.get_metric(_SINGLE_SPAN_SAMPLING_RATE) == sample_rate
+    assert span.get_metric(_SINGLE_SPAN_SAMPLING_MECHANISM) == mechanism
+    assert span.get_metric(_SINGLE_SPAN_SAMPLING_MAX_PER_SEC) == limit
+
+
+def traced_function(rule):
+    tracer = DummyTracer()
+    with tracer.trace("test_name") as span:
+        span.service = "test_service"
+        rule.sample(span)
+    return span
+
+
+def test_single_span_rule_default_no_match():
+    rule = SpanSamplingRule()
+    span = traced_function(rule)
+
+    assert_sampling_decision_tags(span, sample_rate=None, mechanism=None, limit=None)
+
+
+def test_single_span_rule_match():
+    rule = SpanSamplingRule(service="test_service", name="test_name")
+    span = traced_function(rule)
+
+    assert_sampling_decision_tags(span)
+
+
+def test_single_span_rule_match_max_per_sec():
+    rule = SpanSamplingRule(service="test_service", name="test_name", max_per_second=133)
```

Comment:
This test should check if such a rule marks a span as single span sampled only after a rate limit threshold is passed.